### PR TITLE
Improve CLI code and descriptions

### LINKED ISSUE #28444: cli.HelpFlag = nil can be used now when update urfave/cli to v2.26.0
### Description

In fact, it's not a bug but rather a TODO that needs attention. I've noticed a TODO at cmd/main.go:L53. Fortunately, the related issue on urfave/cli has been resolved in version v2.26.0. Therefore, it should be safe for us to update the version and comment out the TODO.

The issue was tracked here: [urfave/cli/issues/1794](https://github.com/urfave/cli/issues/1794), and the fix is included in the release: [v2.26.0](https://github.com/urfave/cli/releases/tag/v2.26.0).

If you find this proposal suitable, I'll proceed with a minor fix.

### Gitea Version

main

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Actually I'm not running Gitea.

### Database

None

### PR DESCRIPTION
**Pull request by wxiaoguang**
* Close #28444 
  * Actually, it doesn't need to use that trick because it looks like it is not necessary, no user really needs it
* Remove the hidden (legacy) "doctor" subcommand and update documents
* Fix "actions" usage

image

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -121,8 +115,9 @@ func prepareWorkPathAndCustomConf(action cli.ActionFunc) func(ctx *cli.Context)
 func NewMainApp(version, versionExtra string) *cli.App {
 	app := cli.NewApp()
 	app.Name = "Gitea"
+	app.HelpName = "gitea"
 	app.Usage = "A painless self-hosted Git service"
-	app.Description = `By default, Gitea will start serving using the web-server with no argument, which can alternatively be run by running the subcommand "web".`
+	app.Description = `Gitea program contains "web" and other subcommands. If no subcommand is given, it starts the web server by default. Use "web" subcommand for more web server arguments, user other subcommands for other features.`
```
Comment (delvh) [inline cmd/main.go]:
```suggestion
	app.Description = `The gitea binary contains multiple subcommands, among them "web" . If no subcommand is given, the web server is started. Use the "web" subcommand to configure how the web server is started, or another subcommand for other purposes.`
```

**Review by delvh — APPROVED** [review]

File: cmd/main.go (hunk 1 above)
Comment (wxiaoguang) [reply cmd/main.go]: I am not a fan of using "the" everywhere.

See the screenshot and old sentences, we never say "the Gitea web server", "Dump the Gitea files", "output the CLI document".


> When to Use 'A,' 'An,' or 'The'

https://gallaudet.edu/student-success/tutorial-center/english-center/grammar-and-vocabulary/definite-and-indefinite-articles/when-to-use-a-an-or-the/

Comment (wxiaoguang) [reply cmd/main.go]: The reason that I think we should put "web" subcommand before others: it is the default one and is the only one most users really need. 

I am not a native English speaks but `among them "web"` reads strange to me.

File: cmd/actions.go [hunk 2]
```diff
@@ -15,9 +15,8 @@ import (
 var (
 	// CmdActions represents the available actions sub-commands.
 	CmdActions = &cli.Command{
-		Name:        "actions",
-		Usage:       "",
-		Description: "Commands for managing Gitea Actions",
+		Name:  "actions",
+		Usage: "Manage Gitea Actions",
```
Comment (silverwind) [inline cmd/actions.go:19]: Go formatting is so weird, why does this de-indent now? 😉

Comment (wxiaoguang) [reply cmd/actions.go:19]:
```
Short1:      1,
Short2:      2,
Loooooooong: 3,
```

Then

```
Short1: 1,
Short2: 2,
Short3: 3,
```

Comment (silverwind) [reply cmd/actions.go:19]: Yes, but the `Subcommands` below should cause indentation but doesn't. Seems functions are exempt from indenation.

Comment (wxiaoguang) [reply cmd/actions.go:19]: Yup, IIRC complex values exempt from "indentation rules"

**Review by lunny — APPROVED** [review]

**Pull request merged**